op-supervisor/supervisor/backend/backend.go

```diff
@@ -423,94 +423,94 @@ func (su *SupervisorBackend) DependencySet() depset.DependencySet {
 // Query methods
 // ----------------------------
 
-func (su *SupervisorBackend) CheckMessage(ctx context.Context, identifier types.Identifier, payloadHash common.Hash, executingDescriptor types.ExecutingDescriptor) (types.SafetyLevel, error) {
-	logHash := types.PayloadHashToLogHash(payloadHash, identifier.Origin)
-	chainID := identifier.ChainID
-	blockNum := identifier.BlockNumber
-	logIdx := identifier.LogIndex
-	_, err := su.chainDBs.Contains(chainID,
-		types.ContainsQuery{
-			BlockNum:  blockNum,
-			Timestamp: identifier.Timestamp,
-			LogIdx:    logIdx,
-			LogHash:   logHash,
-		})
-	if errors.Is(err, types.ErrFuture) {
-		su.logger.Debug("Future message", "identifier", identifier, "payloadHash", payloadHash, "err", err)
-		return types.LocalUnsafe, nil
+func (su *SupervisorBackend) checkAccess(ctx context.Context,
+	acc types.Access, executingDescriptor types.ExecutingDescriptor) (eth.BlockID, error) {
+
+	// Check upper-bound invariant, strictly
+	// (for access-lists we don't afford to check intra-timestamp dependencies)
+	if executingDescriptor.Timestamp <= acc.Timestamp {
+		// err
 	}
-	if errors.Is(err, types.ErrConflict) {
-		su.logger.Debug("Conflicting message", "identifier", identifier, "payloadHash", payloadHash, "err", err)
-		return types.Invalid, nil
+
+	// Check message expiry
+	expiryWindow := su.depSet.MessageExpiryWindow()
+	expiryAt := acc.Timestamp + expiryWindow
+	if expiryAt < acc.Timestamp {
+		// overflow
 	}
-	if err != nil {
-		return types.Invalid, fmt.Errorf("failed to check log: %w", err)
+	if executingDescriptor.Timestamp > expiryAt {
+		// err
 	}
-	if identifier.Timestamp+su.depSet.MessageExpiryWindow() < executingDescriptor.Timestamp {
-		su.logger.Debug("Message expired", "identifier", identifier, "payloadHash", payloadHash, "executingTimestamp", executingDescriptor.Timestamp)
-		return types.Invalid, nil
+	// If a timeout is set, check if executing late is still within the expiry window
+	if executingDescriptor.Timeout != nil {
+		timeout := *executingDescriptor.Timeout
+		if executingDescriptor.Timestamp+timeout < executingDescriptor.Timestamp {
+			// overflow err
+		}
+		if executingDescriptor.Timestamp+timeout < expiryAt {
+			// expiry err
+		}
```

Comment:
Timestamp checks are getting large enough that maybe they should be abstracted.

Reply:
refactored it to be a helper method attached to the executing-descriptor type